xdsl/dialects/riscv.py

```diff
@@ -57,6 +59,11 @@ class Register:
     name: str | None = field(default=None)
     """The register name. Should be one of `ABI_INDEX_BY_NAME` or `None`"""
 
+    ABI_INDEX_BY_NAME: dict[str, int] = field(init=False)
+
+
+@dataclass(frozen=True)
+class IntegerRegister(Register):
```

Comment:
Do you think it might be better to call this `IntRegister` instead? I'm usually all for full length names of things but it's just used so often that I don't think it's that useful to have the extra `eger` everywhere. This is more of a nit.

Reply:
Renamed.